superbench/common/utils/topo_aware.py

```diff
@@ -35,37 +35,23 @@ def search(self, pattern, string, flags=0):
         return self.matched
 
 
-def gen_ibstat_file(ibstat_file):
-    """Generate ibstat file for each node with specified path.
+def gen_ibstat_file(host_list, ibstat_file):
+    """Generate ibstat file in each node with specified path.
 
     Args:
+        host_list (list): list of VM read from hostfile.
         ibstat_file (str): path of ibstat output.
     """
-    from mpi4py import MPI
-
-    if not MPI.Is_initialized():
-        MPI.Init()
-
-    comm = MPI.COMM_WORLD
-    name = MPI.Get_processor_name()
-
-    # The command to fetch ibstat info
-    cmd = r"ibstat | grep -Po 'System image GUID: \K\S+$'"
-    output = os.popen(cmd)
-    ibstat = 'VM_hostname ' + name + '\n' + str(output.read())
-
-    # Fetch all ibstate from each node
-    ibstats = comm.allgather(ibstat)
-
-    ibstate_file_path = Path(ibstat_file)
-
-    # Filter the duplicate info
-    ibstat_infos = set(ibstats)
-
-    with ibstate_file_path.open(mode='w') as f:
-        for ibstat_info in ibstat_infos:
-            f.write(ibstat_info)
-    MPI.Finalize()
+    try:
+        pssh_cmd = "pssh -i -t 5 -H '{}' ".format(' '.join(host_list))
+        cmd = "'cat /sys/class/infiniband/*/sys_image_guid | tr -d :' | sed 's/^.*SUCCESS]/VM_hostname/g' | uniq"
```

Comment:
pls remove uniq, it's useless if there's no sort ahead
what if failed vm also prints some lines, it will be mixed with success output?

Reply:
> pls remove uniq, it's useless if there's no sort ahead
sure, will address
> what if failed vm also prints some lines, it will be mixed with success output?
can we add an error log here if the output contains the fail case?

Reply:
maybe check "FAILURE" line when parsing the log and ignore following lines until next VM_hostname? you may enhance it in the future instead of this fix

here's a case:
```
VM_hostname node0
000100020003000a
000100020003000b
[2] 03:26:16 [FAILURE] node1 Exited with error code 255
anything in error log
VM_hostname node2
000100020003000d
000100020003000c
```